x/evm/statedb/state_object.go

```diff
@@ -0,0 +1,245 @@
+package statedb
+
+import (
+	"bytes"
+	"math/big"
+	"sort"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+var emptyCodeHash = crypto.Keccak256(nil)
+
+// Account is the Ethereum consensus representation of accounts.
+// These objects are stored in the main account trie.
+type Account struct {
+	Nonce    uint64
+	Balance  *big.Int
+	CodeHash []byte
+}
+
+// NewEmptyAccount returns an empty account.
+func NewEmptyAccount() *Account {
+	return &Account{
+		Balance:  new(big.Int),
+		CodeHash: emptyCodeHash,
+	}
+}
+
+// IsContract returns if the account contains contract code.
+func (acct Account) IsContract() bool {
+	return !bytes.Equal(acct.CodeHash, emptyCodeHash)
+}
+
+type Storage map[common.Hash]common.Hash
```

Comment:
ditto move to `types/` and add comment

Reply:
- https://github.com/tharsis/ethermint/pull/729#discussion_r778501094
- added a comment.